src/pardibaal/DBM.cpp

```diff
@@ -112,6 +110,10 @@ namespace pardibaal {
     relation_t DBM::relation(const DBM &dbm) const {
         if (this->dimension() != dbm.dimension())
             return relation_t::different();
+        else if (this->is_empty())
```

Comment:
Probably you only want to do the cached version here.
Relation is n^2, so in case one of them does not have the cached result, just do the relation?